src/main/java/ch/njol/skript/lang/util/SimpleExpression.java

```diff
@@ -249,45 +250,63 @@ public void change(final Event e, final @Nullable Object[] delta, final ChangeMo
 	 * @see #setTime(int, Expression, Class...)
 	 */
 	@Override
-	public boolean setTime(final int time) {
+	public boolean setTime(int time) {
 		if (getParser().getHasDelayBefore() == Kleenean.TRUE && time != 0) {
-			Skript.error("Can't use time states after the event has already passed");
+			Skript.error("Can't use time states after the event has already passed.");
 			return false;
 		}
 		this.time = time;
 		return false;
 	}
-	
-	protected final boolean setTime(final int time, final Class<? extends Event> applicableEvent) {
+
+	protected final boolean setTime(int time, Class<? extends Event> applicableEvent) {
 		if (getParser().getHasDelayBefore() == Kleenean.TRUE && time != 0) {
-			Skript.error("Can't use time states after the event has already passed");
+			Skript.error("Can't use time states after the event has already passed.");
 			return false;
 		}
 		if (!getParser().isCurrentEvent(applicableEvent))
 			return false;
 		this.time = time;
 		return true;
 	}
-	
-	protected final boolean setTime(final int time, final Class<? extends Event> applicableEvent, final Expression<?>... mustbeDefaultVars) {
+
+	@SafeVarargs
+	protected final boolean setTime(int time, Class<? extends Event>... applicableEvents) {
+		if (getParser().getHasDelayBefore() == Kleenean.TRUE && time != 0) {
+			Skript.error("Can't use time states after the event has already passed.");
+			return false;
+		}
+		if (!getParser().isCurrentEvent(applicableEvents))
+			return false;
+		this.time = time;
+		return true;
+	}
+
+	protected final boolean setTime(int time, Class<? extends Event> applicableEvent, @NonNull Expression<?>... mustbeDefaultVars) {
 		if (getParser().getHasDelayBefore() == Kleenean.TRUE && time != 0) {
-			Skript.error("Can't use time states after the event has already passed");
+			Skript.error("Can't use time states after the event has already passed.");
 			return false;
 		}
 		if (!getParser().isCurrentEvent(applicableEvent))
 			return false;
-		for (final Expression<?> var : mustbeDefaultVars) {
+		for (Expression<?> var : mustbeDefaultVars) {
 			if (!var.isDefault()) {
 				return false;
 			}
 		}
 		this.time = time;
 		return true;
 	}
-	
-	protected final boolean setTime(final int time, final Expression<?> mustbeDefaultVar, final Class<? extends Event>... applicableEvents) {
+
+	@SafeVarargs
+	protected final boolean setTime(int time, Expression<?> mustbeDefaultVar, Class<? extends Event>... applicableEvents) {
 		if (getParser().getHasDelayBefore() == Kleenean.TRUE && time != 0) {
-			Skript.error("Can't use time states after the event has already passed");
+			Skript.error("Can't use time states after the event has already passed.");
+			return false;
+		}
+		if (mustbeDefaultVar == null) {
```

Comment:
mustBeDefaultVar is annotated as non null though - is there somewhere that violates this?